modules/ui/fields/combo.js

```diff
@@ -410,11 +417,12 @@ export function uiFieldCombo(field, context) {
         if (_isMulti) {
             t[d.key] = undefined;
         } else if (_isSemi) {
-            var arr = _multiData.map(function(md) {
-                return md.key === d.key ? null : md.key;
-            }).filter(Boolean);
+            let arr = _multiData.map(item => item.key);
+            arr.splice(d.index, 1);
```

Comment:
Maybe add a comment here that this is the place where we actually delete the entry. I find the splice to be less obvious than the filtering that was used before in signaling that this is delete action.

Reply:
good idea, done